go/libraries/doltcore/sqle/enginetest/dolt_queries_workspace.go

```diff
@@ -333,4 +332,633 @@ var DoltWorkspaceScriptTests = []queries.ScriptTest{
 			*/
 		},
 	},
+	{
+		Name: "dolt_workspace_* prevent illegal updates",
+		SetUpScript: []string{
+			"create table tbl (pk int primary key, val int);",
+			"insert into tbl values (42,42);",
+			"call dolt_commit('-Am', 'creating table tbl');",
+			"update tbl set val=51 where pk=42;",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:          "update dolt_workspace_tbl set to_val = 108 where id = 0;",
+				ExpectedErrStr: "only update of column 'staged' is allowed",
+			},
+		},
+	},
+	{
+		Name: "dolt_workspace_* modifies promote to staging",
+		SetUpScript: []string{
+			"create table tbl (pk int primary key, x int, y int);",
+			"insert into tbl values (41,42,43);",
+			"insert into tbl values (50,51,52);",
+			"call dolt_commit('-Am', 'creating table tbl');",
+			"update tbl set x=23",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query: "select * from dolt_workspace_tbl",
+				Expected: []sql.Row{
+					{0, false, "modified", 41, 23, 43, 41, 42, 43},
+					{1, false, "modified", 50, 23, 52, 50, 51, 52},
+				},
+			},
+			{
+				Query: "update dolt_workspace_tbl set staged = true where to_pk = 41;",
+			},
+			{
+				Query: "select * from dolt_workspace_tbl",
+				Expected: []sql.Row{
+					{0, true, "modified", 41, 23, 43, 41, 42, 43},
+					{1, false, "modified", 50, 23, 52, 50, 51, 52},
+				},
+			},
+			{
+				Query: "update dolt_workspace_tbl set staged = 1 where staged = 0;",
+			},
+			{
+				Query: "select * from dolt_workspace_tbl",
+				Expected: []sql.Row{
+					{0, true, "modified", 41, 23, 43, 41, 42, 43},
+					{1, true, "modified", 50, 23, 52, 50, 51, 52},
+				},
+			},
+			{
+				Query: "update tbl set y=81",
+			},
+			{
+				Query: "select * from dolt_workspace_tbl",
+				Expected: []sql.Row{
+					{0, true, "modified", 41, 23, 43, 41, 42, 43},
+					{1, true, "modified", 50, 23, 52, 50, 51, 52},
+					{2, false, "modified", 41, 23, 81, 41, 23, 43},
+					{3, false, "modified", 50, 23, 81, 50, 23, 52},
+				},
+			},
```

Comment:
(minor) might be nice to test the staged/unstaged-ness of the rows through another way, like dolt_diff `WORKING` / `STAGED` or something.

Reply:
Added. Also, we have a few other tests which do similar checks.